ts/IterableInboxMessageList.tsx

```diff
@@ -1,17 +1,26 @@
-'use strict'
-import React, { ReactElement } from 'react'
-import { ScrollView, Text, StyleSheet } from 'react-native'
-import IterableInboxMessageCell from './IterableInboxMessageCell'
+'use strict';
+import React, { ReactElement } from 'react';
+import { ScrollView, Text, StyleSheet } from 'react-native';
+import IterableInboxMessageCell from './IterableInboxMessageCell';
 
 type MessageListProps = {
-   messages: Array<any>  
+   messages: Array<any>,
+   handleMessageSelect: Function  
 }
 
-const IterableInboxMessageList = ({ messages }: MessageListProps) => {
+const IterableInboxMessageList = ({ messages, handleMessageSelect }: MessageListProps) => {
    function flaggedMessage(message: { [key: string]: any }, index: number) {
       return (index === messages.length - 1) ?
-         <IterableInboxMessageCell message={message} last={true} /> :
-         <IterableInboxMessageCell message={message} last={false} />
+         <IterableInboxMessageCell
+            index={index} 
+            message={message}
+            handleMessageSelect={(index: number) => handleMessageSelect(index, messages)} 
+            last={true} /> :
```

Comment:
A smaller version of the refactoring idea here; the only difference here is the `last` being `true` or `false`, otherwise the code is the same, but duplicated.

Reply:
Yes I agree. It would be better to send down a flag that represents if the message is the last message on one message cell component. I will make the update.